nbconvert/filters/markdown_mistune.py

```diff
@@ -21,15 +21,58 @@
 from nbconvert.filters.strings import add_anchor
 
 
+class MathBlockGrammar(mistune.BlockGrammar):
+    """This defines a single regex comprised of the different patterns that 
+    identify math content spanning multiple lines. These are used by the 
+    MathBlockLexer.
+    """
+    multi_math_str = "|".join([r"(^\$\$.*?\$\$)",
```

Comment:
Do you know about the `re.VERBOSE` option? It lets you split a regex over several lines and have comments, without needing to assemble the string yourself.

Reply:
Could you give an example in this case? 

Because I genuinely prefer reading individual regexes joined with a pipe to indicate alternations to one long regex that is individually difficult to parse.

Reply:
Something like this:

```python
multiline_math = re.compile(r"""(^\$\$.*?\$\$)|
                                (^\\\\\[.*?\\\\\])|
                                (^\\begin\{([a-z]*\*?)\}(.*?)\\end\{\4\})""",
                            re.DOTALL | re.VERBOSE)
```

Personally I prefer that to the string-joining approach, but you're the one writing it, so you get to decide on code style.

Reply:
Interesting, in this case I find them equally easy to read. it's more in cases like this in mistune https://github.com/lepture/mistune/blob/92b7f32664bad1a4b3740ee81eda47e5246e780f/mistune.py#L120-L134 where individual chunks need to split multiple lines, and accordingly detecting that an alternation is happening is a lot harder.

Reply:
Since I find these to be equally readable, I'm going to go with your style in this case. But I also figured out a way to simplify it further (since we don't really need the groups for grabbing the content, as you pointed out).

Reply:
can I ask your rationale for disliking the string joining approach?

Reply:
Ok… actually I'm going to keep using the join method, but with simpler component strings. 

I ended up not finding the pipe's use for alternations and the bitwise OR to be equally readable as not needing the second flag. I had skimmed over that detail at a first glance.